components/style/values/animated/effects.rs

```diff
@@ -0,0 +1,202 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Computed types for CSS values related to effects.
+
+use properties::animated_properties::Animatable;
+#[cfg(feature = "gecko")]
+use properties::animated_properties::IntermediateColor;
+use values::computed::{Angle, Number};
+use values::computed::effects::DropShadow as ComputedDropShadow;
+use values::computed::effects::Filter as ComputedFilter;
+use values::computed::effects::FilterList as ComputedFilterList;
+use values::computed::length::Length;
+use values::generics::effects::Filter as GenericFilter;
+use values::generics::effects::FilterList as GenericFilterList;
+
+/// An animated value for the `filter` property.
+pub type FilterList = GenericFilterList<Filter>;
+
+/// An animated value for a single `filter`.
+pub type Filter = GenericFilter<
+    Angle,
+    // Should be `NumberOrPercentage`.
```

Comment:
And, perhaps worth to be explicit and mention `<number-percentage>`, since currently `NumberOrPercentage` isn't exactly that afaict.